Automattic/wp-calypso
# Subscription-based premium themes: Support multiple premium theme products

#### Proposed Changes

Currently Calypso assumes all premium themes have the same product slug.  But to support having subscription-based themes with different products for each theme (an implementation we are currently investigating) it is necessary to make Calypso check the product type rather than the product slug to determine if a purchase is a premium theme.

This pull request accomplishes that. Props to @mreishus for part of the code here (from https://github.com/Automattic/wp-calypso/pull/66319).

Related to: paYKcK-1Qr-p2

#### Testing Instructions

The most important testing is to buy a premium theme via `/themes/premium/[YOURSITE]` as a one-time purchase and make sure the entire experience is unchanged by this pull request:
1. Buy the theme and make sure it can be activated during the purchase process.
2. Go back to `/themes/premium/[YOURSITE]` (with the purchased theme either set as active or not) and make sure it displays correctly.
3. Go to `/themes/premium/[YOURSITE]?flags=signup/seller-upgrade-modal` (per https://github.com/Automattic/wp-calypso/pull/66285#issuecomment-1206744656) and make sure it looks correct there too. Note that it looks like the theme is being misidentified as an annual subscription on this page (even though it's a one-time purchase) but that is a preexisting issue.
4. Go to `/setup/designSetup?siteSlug=[YOURSITE]&flags=signup/design-picker-unified` (per https://github.com/Automattic/wp-calypso/pull/66319) and make sure it looks correct there too. As with the above, it looks like the theme is being misidentified as an annual subscription on this page (even though it's a one-time purchase) but that is a preexisting issue.
5. Cancel/refund the theme and make sure that works correctly also.

You can also test with D85381-code on the server and follow the instructions there to test purchasing an annual subscription. Following similar testing steps as above, the theme should display in a reasonable way everywhere in the user interface.

#### Pre-merge Checklist
